Kernel/LPO.hpp

```diff
@@ -43,21 +45,24 @@ class LPO
   ~LPO() override = default;
 
   using PrecedenceOrdering::compare;
-  [[nodiscard]] Result compare(TermList tl1, TermList tl2) const override;
+  Result compare(TermList tl1, TermList tl2) const override;
```

Comment:
Could these be `[[nodiscard]]` again?

Reply:
I'm a bit confused because I get `-Wunused-result` warnings even without `[[nodiscard]]`. Do we need this across all compilers?

Reply:
Ahah - `Result` itself is now `[[nodiscard]]`, so these are useless. Thanks!